src/tests/ftest/security/pool_security_acl.py

```diff
@@ -0,0 +1,400 @@
+'''
+from __future__ import print_function
+
+import os
+from apricot import TestWithServers
+from daos_utils import DaosCommand
+from dmg_utils import DmgCommand
+import dmg_utils
+import random
+import pwd
+import grp
+import re
+from general_utils import pcmd
+
+
+PERMISSIONS = ["", "r", "w", "rw"]
+
+def acl_entry(usergroup, name, permission):
+    '''
+    Deascription:
+        Create a daos acl entry on the specified user or groups
+        with specified or random permission.
+    Args:
+        usergroup: user or group.
+        name: user or group name to be created.
+        permission: permission to be created.
+    Return:
+        entry: daos acl entry.
+    '''
+    if PERMISSIONS == "random":
+        permission = PERMISSIONS[random.randint(0, 3)]
+    if permission == "nonexist":
+        return ""
+    if "group" in usergroup:
+        entry = "A:G:" + name + "@:" + permission
+    else:
+        entry = "A::" + name + "@:" + permission
+    return entry
+
+def add_del_user(hosts, ba_cmd, user):
+    '''
+    Deascription:
+        Add or delete the daos user and group on host by sudo command.
+    Args:
+        hosts: list of host.
+        ba_cmd: linux bash command to create user or group.
+        user: user or group name to be created or cleaned.
+    Return:
+        none.
+    '''
+    bash_cmd = os.path.join("/usr/sbin", ba_cmd)
+    homedir = ""
+    if "user" in ba_cmd:
+        homedir = "-r"
+    cmd = " ".join(("sudo", bash_cmd, homedir, user))
+    pcmd(hosts, cmd, False)
+
+class PoolSecurityTest(TestWithServers):
+    '''
+    Test Class Description:
+    Tests to verify the Pool security with dmg and daos tools on daos_server.
+    :avocado: recursive
+    '''
+
+    def get_pool_acl_list(self, uuid):
+        '''
+        Deascription:
+            Get daos pool acl list by dmg get-acl
+        Args:
+            uuid: pool uuid number.
+        Return:
+            pool_permission_list: daos pool acl list.
+        '''
+        dmg = DmgCommand(os.path.join(self.prefix, "bin"))
+        dmg.request.value = "pool"
+        dmg.action.value = "get-acl --pool " + uuid
+        port = self.params.get("port", "/run/server_config/*")
+        servers_with_ports = [
+            "{}:{}".format(host, port) for host in self.hostlist_servers]
+        dmg.hostlist.update(",".join(servers_with_ports), "dmg.hostlist")
+        result = dmg.run()
+
+        pool_permission_list = []
+        for line in result.stdout.splitlines():
+            if not line.startswith("A:"):
+                continue
+            elif line.startswith("A::"):
+                found_user = re.search(r"A::(.+)@:(.*)", line)
+                if found_user:
+                    pool_permission_list.append(line)
+            elif line.startswith("A:G:"):
+                found_group = re.search(r"A:G:(.+)@:(.*)", line)
+                if found_group:
+                    pool_permission_list.append(line)
```

Comment:
All this should be OK for now, but we will have other flags besides Group (G) in the future, and other types of entry as well besides Allow (A).

Reply:
Test script can be expended with the new function become available.
Thanks.